### ISSUE TITLE
[qt5-base:x64-osx] build failure

### ISSUE THREAD

**Issue by cenit** [microsoft/vcpkg#12387]
**Host Environment**
- OS: macOS 10.15.5
- Compiler: 
```
$ clang --version
Apple clang version 11.0.3 (clang-1103.0.32.62)
Target: x86_64-apple-darwin19.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**To Reproduce**
`./vcpkg install qt5-base`

**Failure logs** 
```
Starting package 3/4: qt5-base:x64-osx
Building package qt5-base[core]:x64-osx...
-- Figuring out qt target mkspec. Target arch x64
-- Target mkspec set to: macx-clang
-- Host mkspec not set. Qt's own buildsystem will try to figure out the host system
-- Using cached /Users/ci/vcpkg/downloads/qtbase-everywhere-src-5.15.0.tar.xz
-- Extracting source /Users/ci/vcpkg/downloads/qtbase-everywhere-src-5.15.0.tar.xz
-- Applying patch patches/winmain_pro.patch
-- Applying patch patches/windows_prf.patch
-- Applying patch patches/qt_app.patch
-- Applying patch patches/gui_configure.patch
-- Applying patch patches/icu.patch
-- Applying patch patches/xlib.patch
-- Applying patch patches/egl.patch
-- Applying patch patches/8c44d70.diff
-- Applying patch patches/Qt5BasicConfig.patch
-- Applying patch patches/Qt5PluginTarget.patch
-- Applying patch patches/create_cmake.patch
-- Applying patch patches/Qt5GuiConfigExtras.patch
-- Using source at /Users/ci/vcpkg/buildtrees/qt5-base/src/5.15.0-f256990b37.clean
-- Detected OSX SDK Version: 10.15.4
-- Major.Minor OSX SDK Version: 10.15
-- Enviromnent OSX SDK Version: 10.15
-- Configuring x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: /Users/ci/vcpkg/buildtrees/qt5-base/src/5.15.0-f256990b37.clean/configure -confirm-license -opensource -verbose -system-zlib -system-libjpeg -system-libpng -system-freetype -system-pcre -system-doubleconversion -system-sqlite -system-harfbuzz -icu -no-angle -openssl-linked -fontconfig -nomake examples -nomake tests -static LIBJPEG_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libjpeg.a ZLIB_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libz.a "LIBPNG_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libpng16d.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libz.a" PCRE2_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libpcre2-16.a "FREETYPE_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libfreetyped.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libbz2d.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libpng16d.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libz.a" "ICU_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libicui18n.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libicutu.a  /Users/ci/vcpkg/installed/x64-osx/debug/lib/libicuuc.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libicuio.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libicudata.a" QMAKE_LIBS_PRIVATE+=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libbz2d.a QMAKE_LIBS_PRIVATE+=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libpng16d.a "PSQL_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libpq.dylib /Users/ci/vcpkg/installed/x64-osx/debug/lib/libssl.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libcrypto.a -ldl -lpthread" "SQLITE_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libsqlite3.a -ldl -lpthread" "HARFBUZZ_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libharfbuzz.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libfreetyped.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libbz2d.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libpng16d.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libz.a -framework ApplicationServices" "OPENSSL_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libssl.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libcrypto.a -ldl -lpthread" "FONTCONFIG_LIBS=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libfontconfig.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libfreetyped.a /Users/ci/vcpkg/installed/x64-osx/debug/lib/libexpat.a -liconv" -debug -prefix /Users/ci/vcpkg/installed/x64-osx -extprefix /Users/ci/vcpkg/installed/x64-osx -hostprefix /Users/ci/vcpkg/installed/x64-osx/tools/qt5/debug -hostlibdir /Users/ci/vcpkg/installed/x64-osx/tools/qt5/debug/lib -hostbindir /Users/ci/vcpkg/installed/x64-osx/tools/qt5/debug/bin -archdatadir /Users/ci/vcpkg/installed/x64-osx/tools/qt5/debug -datadir /Users/ci/vcpkg/installed/x64-osx/share/qt5/debug -plugindir /Users/ci/vcpkg/installed/x64-osx/debug/plugins -qmldir /Users/ci/vcpkg/installed/x64-osx/debug/qml -headerdir /Users/ci/vcpkg/installed/x64-osx/include -libexecdir /Users/ci/vcpkg/installed/x64-osx/tools/qt5/debug -bindir /Users/ci/vcpkg/installed/x64-osx/debug/bin -libdir /Users/ci/vcpkg/installed/x64-osx/debug/lib -I /Users/ci/vcpkg/installed/x64-osx/include -L /Users/ci/vcpkg/installed/x64-osx/debug/lib -L /Users/ci/vcpkg/installed/x64-osx/debug/lib/manual-link -xplatform macx-clang
    Working Directory: /Users/ci/vcpkg/buildtrees/qt5-base/x64-osx-dbg
    Error code: 3
    See logs for more information:
      /Users/ci/vcpkg/buildtrees/qt5-base/config-x64-osx-dbg-out.log
      /Users/ci/vcpkg/buildtrees/qt5-base/config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  ports/qt5-base/cmake/configure_qt.cmake:110 (vcpkg_execute_required_process)
  ports/qt5-base/portfile.cmake:292 (configure_qt)
  scripts/ports.cmake:76 (include)


Error: Building package qt5-base:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: qt5-base:x64-osx
  Vcpkg version: 2020.06.15-unknownhash
```

from config-x64-osx-dbg-err.log:
```
Checking for C++17 support... 
+ cd /Users/ci/vcpkg/buildtrees/qt5-base/x64-osx-dbg/config.tests/cxx17 && /Users/ci/vcpkg/buildtrees/qt5-base/x64-osx-dbg/bin/qmake "CONFIG -= qt debug_and_release app_bundle lib_bundle" "CONFIG += static warn_off console single_arch" 'QMAKE_LIBDIR += /Users/ci/vcpkg/installed/x64-osx/debug/lib /Users/ci/vcpkg/installed/x64-osx/debug/lib/manual-link' 'INCLUDEPATH += /Users/ci/vcpkg/installed/x64-osx/include' QMAKE_LIBS_PRIVATE+=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libbz2d.a QMAKE_LIBS_PRIVATE+=/Users/ci/vcpkg/installed/x64-osx/debug/lib/libpng16d.a /Users/ci/vcpkg/buildtrees/qt5-base/x64-osx-dbg/config.tests/cxx17
+ cd /Users/ci/vcpkg/buildtrees/qt5-base/x64-osx-dbg/config.tests/cxx17 && MAKEFLAGS= /usr/bin/make
> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -c -pipe -stdlib=libc++ -O2 -std=gnu++1z  -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.13 -w -fPIC  -I. -I/Users/ci/vcpkg/installed/x64-osx/include -I/Users/ci/vcpkg/buildtrees/qt5-base/src/5.15.0-f256990b37.clean/mkspecs/macx-clang -o main.o main.cpp
> main.cpp:14:18: error: 'get<int, int>' is unavailable: introduced in macOS 10.14
>     int i = std::get<int>(v);
>                  ^
> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/variant:1376:16: note: 'get<int, int>' has been explicitly marked unavailable here
> constexpr _Tp& get(variant<_Types...>& __v) {
>                ^
> main.cpp:15:10: error: 'visit<(lambda at main.cpp:15:16), std::__1::variant<int> &>' is unavailable: introduced in macOS 10.14
>     std::visit([](const auto &) { return 1; }, v);
>          ^
> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/variant:1534:26: note: 'visit<(lambda at main.cpp:15:16), std::__1::variant<int> &>' has been explicitly marked unavailable here
> constexpr decltype(auto) visit(_Visitor&& __visitor, _Vs&&... __vs) {
>                          ^
> 2 errors generated.
> make: *** [main.o] Error 1
test config.qtbase.tests.cxx17 FAILED
``

**Comment by Neumann-A** [comment]
Might be related https://codereview.qt.nokia.com/c/qt/qtbase/+/278068

**Comment by Neumann-A** [comment]
try changing the second argument in from 10.12:
https://github.com/microsoft/vcpkg/blob/0523b5eb57341e44fc0551900e5803b13686821a/ports/qt5-base/portfile.cmake#L264
to `QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.13`

**Comment by JackBoosY** [comment]
Confirmed, but the error message on my machine is different:
```
  1 In file included from /Users/vcpkg/Documents/vcpkg/vcpkg/buildtrees/qt5-base/src/5.15.0-f256990b37.clean/src/corelib/global/qglobal.cpp:97:
  2 In file included from /Users/vcpkg/Documents/vcpkg/vcpkg/buildtrees/qt5-base/src/5.15.0-f256990b37.clean/include/QtCore/5.15.0/QtCore/private/qcore_mac_p.h:1:
  3 In file included from /Users/vcpkg/Documents/vcpkg/vcpkg/buildtrees/qt5-base/src/5.15.0-f256990b37.clean/include/QtCore/5.15.0/QtCore/private/../../../../../src/corelib/kernel/qcore_mac_p.h:63:
  4 In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
  5 In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:23:
  6 In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
  7 In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:208:
  8 In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/HFSVolumes.h:25:
  9 /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/hfs/hfs_format.h:794:2: error: unknown type name 'uuid_string_t'; did you mean 'io_string_t'?
```

**Comment by Neumann-A** [comment]
@JackBoosY: Your issue is probably unrelated to qt
https://github.com/koekeishiya/homebrew-formulae/issues/9

**Comment by cenit** [comment]
in #12229 I had to fix this problem in order to be able to continue working on the PR.

**Comment by Neumann-A** [comment]
@cenit: Qt on other platforms than windows probably also requires a fontconfig update to even be able to display text correctly. (Or a properly setup fontconfig in vcpkg. )

**Comment by Neumann-A** [comment]
@JackBoosY: Issue was closed by #12342
